### ISSUE TITLE
`error: failed to create directory %{_topdir}: /root/rpmbuild: File exists` for RPM

### ISSUE THREAD

**Issue by armab** [StackStorm/st2-packages#171]
I get this error probably 4th time already. It happens randomly, rarely and only for `rpm` builds. I guess it started to appear after removing `st2` components and Rake refactoring.

The log:
```sh
[wheelhouse: st2mistral]      touch .stamp-wheelhouse
[wheelhouse: st2mistral]  Finished in 26.438 seconds with exit status 0 (successful).
[package: st2]  Running /usr/bin/env make changelog as root@centos6build
[package: st2]      make: Nothing to be done for `changelog'.
[package: st2]  Finished in 0.015 seconds with exit status 0 (successful).
[package: st2]  Running /usr/bin/env bash $BASEDIR/scripts/build_os_package.sh st2 as root@centos6build
[package: st2mistral]  Running /usr/bin/env make changelog as root@centos6build
[package: st2mistral]      make: Nothing to be done for `changelog'.
[package: st2mistral]  Finished in 0.012 seconds with exit status 0 (successful).
[package: st2mistral]  Running /usr/bin/env bash $BASEDIR/scripts/build_os_package.sh st2mistral as root@centos6build
[package: st2mistral]      

.circle/docker-compose.sh build ${DISTRO} returned exit code 1

[package: st2mistral] error: failed to create directory %{_topdir}: /root/rpmbuild: File exists Action failed: .circle/docker-compose.sh build ${DISTRO}
```

Considering that sometimes the error appears as:
```
[package: st2mistral] error: failed to create directory
```
and sometimes as:
```
[package: st2] error: failed to create directory
```
I guess it's a race condition somewhere for these packages with directory creation.

![ijmy9lw 1](https://cloud.githubusercontent.com/assets/1533818/13058629/105ac276-d42d-11e5-9076-bb00fbf1cc6e.jpg)

See:
https://circleci.com/gh/StackStorm/st2-packages/422 - `el6`
https://circleci.com/gh/StackStorm/st2-packages/423 - `el7`
for the full log.

**Comment by armab** [comment]
That race still happens sometimes: https://circleci.com/gh/StackStorm/st2/757

**Comment by armab** [comment]
Again: https://stackstorm.slack.com/archives/packages/p1456905313003515

**Comment by armab** [comment]
This error hits us every day, with pretty good stability.

General thoughts.
From one side it's good to fix any bugs which make CI `RED` ASAP,
from the other side implementing `CI for Mistral` will eliminate this bug automatically: https://github.com/StackStorm/st2-packages/issues/82

**Comment by dennybaa** [comment]
you could give it a shot though)

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/stackstorm/p1457702245007228

**Comment by dennybaa** [comment]
resolved in #246
